### ISSUE TITLE
Unable to find ffmpeg

### ISSUE THREAD

**Issue by junkman690** [joshuaboniface/rffmpeg#25]
Hi, I am trying to utilse rffmpeg from inside a jellyfin docker instance to a bare metal transcoding machine running Ubunut22.04, however Jellyfin is unable to pickup the binary. I have double checked the paths, added just the folder without ffmpeg specified and even rebuilt the docker with the same results.

![image](https://user-images.githubusercontent.com/3007893/185028826-7e69074f-c7c1-4a19-a219-b44129559b1b.png)

![image](https://user-images.githubusercontent.com/3007893/185028865-c6185e7a-a85d-4e04-985f-79738817945d.png)

![image](https://user-images.githubusercontent.com/3007893/185028962-6bd7963d-d621-4b16-89b2-6bebb6ffbbf6.png)

**Comment by joshuaboniface** [comment]
First I'd say try putting it right in `/` and seeing if that works. Might be some strange interaction with overlay filesystems.

If that doesn't work, I'd be curious what Jellyfin's debug logs say when you try to save the config. They might provide an error or clue.

**Comment by junkman690** [comment]
Softlinked to /ffmpeg and /ffprobe, same error including trying just /

Log didn't show much [https://pastebin.com/mim09kna](https://pastebin.com/mim09kna)

**Comment by junkman690** [comment]
Ok, I seemed to have been jumping too far ahead and assuming the return from -version was remote, its the local response. SSH is failing out

Initially was failing out:

2022-08-17 15:14:13,103 - rffmpeg[1081] - DEBUG - SSH test command stdout: 
2022-08-17 15:14:13,103 - rffmpeg[1081] - DEBUG - SSH test command stderr: Pseudo-terminal will not be allocated because stdin is not a terminal.

Adding either -t or -T option to the SSH command in rffmpeg.yml got rid of the pseudo-terminal but produced the linked log

[https://pastebin.com/eqVzsqWg](https://pastebin.com/eqVzsqWg)

**Comment by junkman690** [comment]
Ok, SSH issue sorted. The docker instance did not have /run/shm. Creating /run/shm got the remote hosts ffmpeg when running --version. After this folder exists the additional -t is not required.

Jellyfin still complains that it can't find ffmpeg at /usr/local/bin, which is expected as rffmepg was failing back to the local version when SSH wasn't working anyway so Jellyfin would have gotten a response either way. Looks to be a bug in how Jellyfin detects ffmpeg

**Comment by junkman690** [comment]
Its a container permission issue. I am using linuxserver.io's Jellyfin container, because I was testing some of the commands as root not as the ABC user to start with I set the wrong permissions on some folders, and the /run/shm option was in the yml file already to customise, I just overlooked it.

After fixing permissions I can set ffmpeg in Jellyfin to the softlinked rffmpeg. I now just need to test the transcoding part.

I plan on adding rffmpeg as a public docker mod after I get it working correctly. Once I have that sorted I'll add a link here for anyone else trying to achieve this.
